src/ophyd_async/epics/motion/motor.py

```diff
@@ -78,14 +76,26 @@ def move(self, new_position: float, timeout: Optional[float] = None):
             raise RuntimeError("Will deadlock run engine if run in a plan")
         call_in_bluesky_event_loop(self._move(new_position), timeout)  # type: ignore
 
-    def set(self, new_position: float, timeout: Optional[float] = None) -> AsyncStatus:
-        watchers: List[Callable] = []
-        coro = asyncio.wait_for(self._move(new_position, watchers), timeout=timeout)
-        return AsyncStatus(coro, watchers)
+    @WatchableAsyncStatus.wrap
+    async def set(self, new_position: float, timeout: float = 0.0):
+        update, move_status = await self._move(new_position)
+        start = time.monotonic()
+        async for current_position in observe_value(
+            self.user_readback, done_status=move_status
```

Comment:
But here the `done_status=move_status` is correct as we are talking to a motor record